test/find_j_moves_test.cpp

```diff
@@ -0,0 +1,166 @@
+#include <map>
+#include <algorithm>
+#include <initializer_list>
+#include <gtest/gtest.h>
+#include <wl/game_state.h>
+#include <wl/find_j_moves.h>
+#include <wl/default_map.h>
+
+namespace wl {
+std::ostream& operator<<(std::ostream& os, JackResource resource) {
+	switch (resource) {
+	case JackResource::CARRIAGE: os << "CARRIAGE"; break;
+	case JackResource::ALLEY: os << "ALLEY"; break;
+	case JackResource::BOAT: os << "BOAT"; break;
+	}
+	return os;
+}
+
+std::ostream& operator<<(std::ostream& os, JackMove move) {
+	os << "JackMove{destination:" << static_cast<size_t>(move.destination());
+	if (const auto cost = move.cost(); cost.has_value()) {
+		os << ",cost:" << *cost;
+	}
+	os << "}";
+	return os;
+}
+} // namespace wl
+
+namespace {
+const wl::MapGraph k_default_map = wl::default_map();
+
+
+void do_normal_moves_test(std::initializer_list<wl::JackMove> expected_moves,
+	                      const wl::GameState& game_state,
+	                      const wl::History& history)
+{
+	// setup a map to keep track of which expected moves we found
+	std::map<wl::JackMove, bool> expected_moves_found;
+	for (const wl::JackMove expected_move : expected_moves) {
+		expected_moves_found.emplace(expected_move, false);
+	}
+	ASSERT_EQ(expected_moves_found.size(), expected_moves.size());
```

Comment:
This one actually checks that the `expected_moves` list is valid, ie. there are no duplicates. If there **were** duplicates, then the `std::map` would only add an entry for the first instance of the duplicate, and our map would have fewer entries than the input list.